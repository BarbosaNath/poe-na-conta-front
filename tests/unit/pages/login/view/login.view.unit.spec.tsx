import React from 'react';
import Login from '@pages/login/view/login.view';
import { render } from '@testing-library/react';

describe('LoginView', () => {
  it('should render Login view correctly', () => {
    const { container } = render(
      <Login
        registerFormProps={{
          isLoading: false,
          userRegisterBody: {
            name: '',
            useremail: '',
            passwordhash: '',
          },
          errorMessage: '',
          handleRegister: vi.fn(),
          handleChange: vi.fn(),
        }}
        loginFormProps={{
          isLoading: false,
          userLoginBody: {
            useremail: '',
            passwordhash: '',
          },
          errorMessage: '',
          handleLogin: vi.fn(),
          handleChange: vi.fn(),
        }}
      />,
    );

    expect(container).toMatchSnapshot();
  });
});
