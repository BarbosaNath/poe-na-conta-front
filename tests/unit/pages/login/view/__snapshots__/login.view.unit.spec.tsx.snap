// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`LoginView > should render Login view correctly 1`] = `
<div>
  <div
    class="lds--card"
  >
    <div
      class="lds--stack lds--stack--mobile lds--stack--space-xxl lds--stack--direction--row lds--stack--align-stretch lds--stack--justify-evenly"
    >
      <form
        class="lds--stack lds--stack--nowrap lds--stack--full-width lds--stack--space-lg lds--stack--direction--column lds--stack--align-stretch lds--stack--justify-start"
      >
        <h3
          class="lds--text lds--text--bold lds--text--md lds--text--secondary lds--text--center"
        >
          Registre-se
        </h3>
        <div
          class="lds--input__wrapper"
        >
          <label
            class="lds--text lds--text--regular lds--text--md lds--text--default"
          >
            Nome
            <input
              class="lds--input__input"
              placeholder="seu nome"
              value=""
            />
          </label>
        </div>
        <div
          class="lds--input__wrapper"
        >
          <label
            class="lds--text lds--text--regular lds--text--md lds--text--default"
          >
            Email
            <input
              class="lds--input__input"
              placeholder="digite seu email"
              value=""
            />
          </label>
        </div>
        <div
          class="lds--input__wrapper"
        >
          <label
            class="lds--text lds--text--regular lds--text--md lds--text--default"
          >
            Senha
            <input
              class="lds--input__input"
              placeholder="crie uma senha"
              value=""
            />
          </label>
        </div>
        <div
          class="lds--button-layout lds--button-layout--column lds--button-layout--full-width lds--button-layout--spacing--xs"
        >
          <button
            class="lds--button lds--button--primary"
            type="submit"
          >
            Submit
          </button>
        </div>
      </form>
      <div
        class="lds--stack lds--stack--nowrap lds--stack--space-lg lds--stack--direction--column lds--stack--align-center lds--stack--justify-center"
      >
        <p
          class="lds--text lds--text--bold lds--text--md lds--text--secondary"
        >
          OU
        </p>
      </div>
      <form
        class="lds--stack lds--stack--nowrap lds--stack--full-width lds--stack--space-lg lds--stack--direction--column lds--stack--align-stretch lds--stack--justify-start"
      >
        <h3
          class="lds--text lds--text--bold lds--text--md lds--text--secondary lds--text--center"
        >
          Fa√ßa login
        </h3>
        <div
          class="lds--input__wrapper"
        >
          <label
            class="lds--text lds--text--regular lds--text--md lds--text--default"
          >
            Email
            <input
              class="lds--input__input"
              name="email"
              placeholder="digite seu email"
              value=""
            />
          </label>
        </div>
        <div
          class="lds--input__wrapper"
        >
          <label
            class="lds--text lds--text--regular lds--text--md lds--text--default"
          >
            Senha
            <input
              class="lds--input__input"
              name="password"
              placeholder="informe sua senha"
              type="password"
              value=""
            />
          </label>
        </div>
        <div
          class="lds--button-layout lds--button-layout--column lds--button-layout--full-width lds--button-layout--spacing--xs"
        >
          <button
            class="lds--button lds--button--primary"
            type="submit"
          >
            Login
          </button>
          <a
            class="lds--button lds--button--link"
          >
            Esqueceu a senha?
          </a>
        </div>
      </form>
    </div>
  </div>
</div>
`;
